### PR TITLE
Spring boot migration

### PR DESCRIPTION
**Pull request by tonydvp**
https://jira.ft.com/browse/MEM-2644

### CODE REVIEW

File: zuora-event-bridge/src/main/resources/application.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      retryBackoff: ${KAFKA_PRODUCER_RETRY_BACKOFF:5s}
+  zipkin:
+    enabled: ${ZIPKIN_ENABLED:false}
+    baseUrl: ${ZIPKIN_BASE_URL:http://localhost:9411}
+    message-timeout: ${ZIPKIN_SAMPLE_RATE:1}
+    sender:
+      type: web
+  sleuth:
+    sampler:
+      probability: 1.0
```
Comment (mihristov) [inline zuora-event-bridge/src/main/resources/application.yaml]: 1.0 means 100%. Either make it 0.1 or remove it. The default is 0.1

File: zuora-event-bridge/src/main/resources/application.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    originSystemId: http://cmdb.ft.com/systems/zuora-event-bridge
+  kafka:
+    bootstrap-servers: ${KAFKA_PRODUCER_BROKER_LIST:['localhost:9092']}
+    producer:
+      socketTimeout: ${KAFKA_PRODUCER_TIMEOUT:5s}
+      retryBackoff: ${KAFKA_PRODUCER_RETRY_BACKOFF:5s}
+  zipkin:
+    enabled: ${ZIPKIN_ENABLED:false}
+    baseUrl: ${ZIPKIN_BASE_URL:http://localhost:9411}
+    message-timeout: ${ZIPKIN_SAMPLE_RATE:1}
```
Comment (mihristov) [inline zuora-event-bridge/src/main/resources/application.yaml]: What's that?

File: zuora-event-bridge/src/main/java/com/ft/membership/zuoraeventbridge/model/PaymentMethodClosedEvent.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.UUID;
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class PaymentMethodClosedEvent {
+
+    private final String closureType;
```
Comment (mihristov) [inline zuora-event-bridge/src/main/java/com/ft/membership/zuoraeventbridge/model/PaymentMethodClosedEvent.java]: No need to write private final. It comes from @Value

File: zuora-event-bridge/src/main/java/com/ft/membership/zuoraeventbridge/controller/ZuoraStateUpdatedCalloutController.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      ZuoraStateUpdatedEvent event = new ZuoraStateUpdatedEvent(userId);
+      UUID messageId = zuoraStateUpdatedService.sendZuoraStateUpdatedEvent(event);
+
+      operation.wasSuccessful()
+          .yielding(USER_ID, userId)
+          .yielding("messageId", messageId)
+          .log();
+      return ResponseEntity.ok(new CalloutSuccessfullyHandledResponse(messageId));
+    } catch (Exception e) {
+      operation.wasFailure().throwingException(e);
```
Comment (mihristov) [inline zuora-event-bridge/src/main/java/com/ft/membership/zuoraeventbridge/controller/ZuoraStateUpdatedCalloutController.java]: You missed the log()

Comment (mihristov) [reply zuora-event-bridge/src/main/java/com/ft/membership/zuoraeventbridge/controller/ZuoraStateUpdatedCalloutController.java]: Also isn't the aspect supposed to log this error?

File: zuora-event-bridge/src/main/resources/application.yaml (hunk 2 above)
Comment (tonydvp) [reply zuora-event-bridge/src/main/resources/application.yaml]: What is what?

Comment (mihristov) [reply zuora-event-bridge/src/main/resources/application.yaml]: What does the `message-timeout` property do and why do we populate it with the sample rate variable?

Comment (tonydvp) [reply zuora-event-bridge/src/main/resources/application.yaml]: Actually, the sample rate var is not even populated, I was using the default value of 1, maybe a copy paste error from another place ! Fixed

File: Dockerfile [hunk 5]
```diff
@@ -8,6 +8,6 @@ WORKDIR /usr/zuora-event-bridge
 
 ADD zuora-event-bridge/target/zuora-event-bridge.jar .
 
-ADD zuora-event-bridge/src/main/resources/config.yaml .
+ADD zuora-event-bridge/src/main/resources/application.yaml .
 
-CMD ["bash", "-c", "java $JAVA_OPTS -DAPP.ENV=${APP_ENV} -jar zuora-event-bridge.jar server config.yaml"]
\ No newline at end of file
+CMD ["bash", "-c", "java $JAVA_OPTS -DAPP.ENV=${APP_ENV} -jar zuora-event-bridge.jar server application.yaml"]
```
Comment (MightyPixel) [inline Dockerfile]: Isn't the convention to have config.yaml? Maybe we can rename the file rather than changing the convention.

Comment (mihristov) [reply Dockerfile]: I think the convention must be driven by the framework we use and the most commonly used name for this file in Spring Boot is application.yaml. Another example of this are the controllers, which in Dropwizard are named resources.

Comment (tonydvp) [reply Dockerfile]: Yes, and it is not even convention, Spring will only search and work for an application.yaml/config.

**Pull request merged**